pkg/minikube/bootstrapper/certs.go

```diff
@@ -545,3 +575,32 @@ func isValid(certPath, keyPath string) bool {
 
 	return true
 }
+
+func isKubeadmCertValid(cmd command.Runner, certPath string) bool {
+	rr, err := cmd.RunCmd(exec.Command("cat", certPath))
+	if err != nil {
+		klog.Infof("failed to read cert file %s: %v", certPath, err)
+		// if reading the cert failed it's likely first start and it doesn't exist yet so mark as valid
+		return true
+	}
+
+	certData, _ := pem.Decode(rr.Stdout.Bytes())
+	if certData == nil {
+		klog.Infof("failed to decode cert file %s", certPath)
+		return false
+	}
+
+	cert, err := x509.ParseCertificate(certData.Bytes)
+	if err != nil {
+		klog.Infof("failed to parse cert file %s: %v\n", certPath, err)
+		return false
+	}
+
+	if cert.NotAfter.Before(time.Now()) {
+		out.WarningT("Certificate {{.certPath}} has expired. Generating a new one...", out.V{"certPath": filepath.Base(certPath)})
```

Comment:
May I suggest moving "Generating a new one" to [the line](https://github.com/kubernetes/minikube/pull/16249/files#diff-c19d80ca90b0f40897c50414dcfd992cf6abeb549015075ea14a06b130c77d49R371) just before the renewal cmd is called. This this function ends up being reused somewhere itll be a bit misleading to say "generating" when its not actually done here (its just a check)

Reply:
Good call, moved